### PR TITLE
Edit project

### PR DESCRIPTION
**Pull request by billadam28**

### CODE REVIEW

File: src/java/CMTJavaClasses/ProjectProcessor.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import org.hibernate.Transaction;
+
+/**
+ *
+ * @author thodo
+ */
+public class ProjectProcessor {
+    private Projects project;
+    String nameOfProject;
+    String acrOfProject;
```
Comment (billadam28) [inline src/java/CMTJavaClasses/ProjectProcessor.java]: no need for the String variables.

File: src/java/CMTJavaClasses/ProjectProcessor.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        //getGroupsQuery = "from Group g order by g.name asc";
+        //getVisitorsNoLeaderQuery = "from Visitor v where group != null";
+    }
+
+    /**
+     * This method checks if the given name exists or not. 
+     * @param nameOfGroup
+     * @return True or False.
+     */
+
```
Comment (billadam28) [inline src/java/CMTJavaClasses/ProjectProcessor.java]: clear the comments.

File: src/java/CMTJavaClasses/ProjectProcessor.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    /**
+     * This method checks if the given name exists or not. 
+     * @param nameOfGroup
+     * @return True or False.
+     */
+    
+    public void editProjectForm (int projectId) {
+        Session session = NewHibernateUtil.getSessionFactory().openSession();
```
Comment (billadam28) [inline src/java/CMTJavaClasses/ProjectProcessor.java]: a better name could be getProjectDetails(). editProjectForm doesnt make sense.

File: src/java/CMTServlets/EditProjectSrvlt.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    protected void processRequest(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        response.setContentType("text/html;charset=UTF-8");
+        HttpSession session = request.getSession(false);
+        
+        if ((session == null) || (session.getAttribute("userId") == null)) {
+            this.getServletConfig().getServletContext().getRequestDispatcher("/index.jsp?noSession=1").forward(request, response);
+        } else {
+            ProjectProcessor projectProcessor = new ProjectProcessor();
+
```
Comment (billadam28) [inline src/java/CMTServlets/EditProjectSrvlt.java]: you should call the getProjectdetails().

File: src/java/CMTServlets/EditProjectSrvlt.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        
+        if ((session == null) || (session.getAttribute("userId") == null)) {
+            this.getServletConfig().getServletContext().getRequestDispatcher("/index.jsp?noSession=1").forward(request, response);
+        } else {
+            ProjectProcessor projectProcessor = new ProjectProcessor();
+            
+            //String selectProject = request.getParameter("selectProject");
+            //int pId = projectProcessor.getProject().getId();
+            //projectProcessor.editProjectForm(Integer.parseInt(selectProject));
+            request.setAttribute("projectProcessor", projectProcessor);
```
Comment (billadam28) [inline src/java/CMTServlets/EditProjectSrvlt.java]: you should pass the project object like projectProcessor.getProject()

**Comment by Thodoris21** [comment]
ok

**Pull request merged**